util.go

```diff
@@ -3,10 +3,31 @@
 package gosnowflake
 
 import (
+	"context"
 	"database/sql/driver"
+	"github.com/google/uuid"
 	"time"
 )
 
+type contextKey string
+
+// snowflakeRequestIDKey is optional context key to specify request id
+const snowflakeRequestIDKey contextKey = "SNOWFLAKE_REQUEST_ID"
+
+// WithRequestID returns a new context with the specified snowflake request id
+func WithRequestID(ctx context.Context, requestID string) context.Context {
+	return context.WithValue(ctx, snowflakeRequestIDKey, requestID)
```

Comment:
Might I ask why not pass in the requestID as a type UUID (correct me if I'm wrong, but I think you can convert a string to UUID type) without having to change the function signature for every function that uses a requestID?

Reply:
According to the REST api documentation, the `requestID` does not have a hard requirement of being a UUID (it's just a string field)
https://docs.snowflake.com/en/user-guide/data-load-snowpipe-rest-apis.html

The upstream callers may also end up stamping in non-UUID request IDs, for example, our Snowflake Golang App server internally doesn't use UUID format. So allowing `string` matches the API doc better and is more flexible for the clients.

Reply:
I see. That makes sense.